src/test/lib/optimizer/strategy/chunk_pruning_rule_test.cpp

```diff
@@ -102,351 +95,386 @@ TEST_F(ChunkPruningRuleTest, SimplePruningTest) {
+  const auto stored_table_node = StoredTableNode::make("compressed");
 
   // clang-format off
-  auto input_lqp =
+  const auto input_lqp =
   PredicateNode::make(greater_than_equals_(lqp_column_(stored_table_node, ColumnID{0}), int64_t{-3'000'000'000}),
     stored_table_node);
   // clang-format on
 
-  auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+  const auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
 
   EXPECT_EQ(actual_lqp, input_lqp);
-  std::vector<ChunkID> expected_chunk_ids = {};
-  std::vector<ChunkID> pruned_chunk_ids = stored_table_node->pruned_chunk_ids();
+  const auto expected_chunk_ids = std::vector<ChunkID>{};
+  EXPECT_EQ(stored_table_node->pruned_chunk_ids(), expected_chunk_ids);
+}
+
+// Predicates that have a subquery as an argument cannot be used for pruning as we do not know the predicate value yet.
+// However, such predicates should not prevent us from pruning by other predicates. The following two test cases ensure
+// correct pruning for predicates with uncorrelated and correlated subqueries.
+TEST_F(ChunkPruningRuleTest, PredicateWithUncorrelatedSubquery) {
+  const auto stored_table_node_1 = StoredTableNode::make("compressed");
+  const auto stored_table_node_2 = StoredTableNode::make("compressed");
+
+  const auto subquery_1 = ProjectionNode::make(expression_vector(value_(1.1)), DummyTableNode::make());
+  const auto subquery_2 = subquery_1->deep_copy();
+  const auto subquery_3 = ProjectionNode::make(expression_vector(value_(3.3)), DummyTableNode::make());
+
+  // clang-format off
+  // SELECT * FROM compressed WHERE a > 200 AND b = (SELECT 1.1);
+  const auto input_lqp_1 =
+  PredicateNode::make(greater_than_(stored_table_node_1->get_column("a"), value_(200)),  // Prunes chunk 1.
+    PredicateNode::make(equals_(stored_table_node_1->get_column("b"), lqp_subquery_(subquery_1)),
+      stored_table_node_1));
+
+  // SELECT * FROM compressed WHERE a > 200 AND b BETWEEN (SELECT 1.1) AND (SELECT 3.3);
+  const auto input_lqp_2 =
+  PredicateNode::make(greater_than_(stored_table_node_2->get_column("a"), value_(200)),  // Prunes chunk 1.
+    PredicateNode::make(between_inclusive_(stored_table_node_2->get_column("b"), lqp_subquery_(subquery_2), lqp_subquery_(subquery_3)),  // NOLINT(whitespace/line_length)
+      stored_table_node_2));
+  // clang-format on
 
-  EXPECT_EQ(pruned_chunk_ids, expected_chunk_ids);
+  const auto actual_lqp_1 = StrategyBaseTest::apply_rule(_rule, input_lqp_1);
+  const auto actual_lqp_2 = StrategyBaseTest::apply_rule(_rule, input_lqp_2);
+
+  EXPECT_EQ(actual_lqp_1, input_lqp_1);
+  EXPECT_EQ(actual_lqp_2, input_lqp_2);
+  const auto expected_chunk_ids = std::vector<ChunkID>{ChunkID{1}};
+  EXPECT_EQ(stored_table_node_1->pruned_chunk_ids(), expected_chunk_ids);
+  EXPECT_EQ(stored_table_node_2->pruned_chunk_ids(), expected_chunk_ids);
+}
+
+TEST_F(ChunkPruningRuleTest, PredicateWithCorrelatedSubquery) {
```

Comment:
Can you add a comment similar to `PredicateWithUncorrelatedSubquery`?

Reply:
I know the comment above states everything. I am just not a fan of reading the comments of other test to understand the current test. Maybe split the comment?

Reply:
We now have
```
// Predicates that have uncorrelated subqueries as arguments cannot be used for pruning as we do not know the predicate
// values yet. However, such predicates should not prevent us from pruning by other predicates.
```

for the first case and
```
// Similar to PredicateWithUncorrelatedSubquery, we cannot use predicates with correlated subqueries as arguments for
// pruning because their values are not known before execution. However, pruning by other predicates is still possible.
```

for the second case.